src/analysis/modifications/generic.jl

```diff
@@ -43,18 +44,20 @@ change_objective(
 
         # Construct objective_indices array
         if typeof(new_objective) == String
-            objective_indices = indexin([new_objective], reactions(model))
+            objective_indices = indexin([new_objective], [reactions(model); genes(model)])
```

Comment:
this will brutally fail for all other models than GeckoModel. Needs to be typed right. I'd vote for a specific version that changes objective towads gene utilization, mixing this with reactions is weird.

Reply:
Hmm, what we actually need is some accessor like: `objective_indices = indexin([new_objective], variables(model))` and the variables can default to `variables(model) = reactions(model)`, but for Gecko `variables(model) = [reactions(model); genes(model)]`. This allows us to extend the change objective thing to other types of models that use, e.g. metabolites as variables too....

Reply:
Nah, the complaint was more like that this should check whether it is working on a model that actually supports the genes BEFORE it does the magic. In particular, the internal function needs to be overloaded.

Reply:
looking at the function, we should also eliminate the union there. :D

Reply:
this will need more work actually, we need to be able to tell which genes are actually variables, it is possible that not all genes are associated with reactions, e.g. if a kcat is not available... I don't think this is possible with gecko at the moment, the exact identity of the genes gets lost. This also means that my mods won't work unless all you care about is minimizing the total amount of enzymes AND all the gene products are used...

Reply:
Seems we did actually do it intelligently, so this was less of a hassle than expected :) However, we still should change "reactions" to "variables" and make "reactions" mean reactions

Reply:
yeah there's the issue for that. :]